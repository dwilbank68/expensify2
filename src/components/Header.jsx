import React, {Component} from 'react';
import {connect} from 'react-redux';
import {NavLink} from 'react-router-dom';

import {startLogout} from "../actions/authActionGenerators";

export
class Header extends Component {

    render() {
        return (
            <header className="header">
                <h1>Expensify</h1>
                <NavLink to="/dashboard" activeClassName="is-active" >Dashboard</NavLink>
                <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
                <button onClick={this.props.startLogout}>
                    Logout
                </button>
            </header>
        );
    }
}

const mapDispatchToProps = dispatch => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(null, mapDispatchToProps)(Header);
