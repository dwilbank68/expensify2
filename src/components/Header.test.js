import React from 'react';

import {shallow} from 'enzyme';
import Header from './Header.jsx';

import '../setupTests.js';

test('should render Header', () => {
    const wrapper = shallow(<Header/>);
    expect(wrapper).toMatchSnapshot();
})