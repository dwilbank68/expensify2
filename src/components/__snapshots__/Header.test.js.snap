// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header 1`] = `
<header
  className="header"
>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/create"
  >
    Create Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/edit"
  >
    Edit Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/help"
  >
    Help
  </NavLink>
  Header
</header>
`;
